pkg/services/ngalert/notifier/channels/pagerduty_test.go

```diff
@@ -0,0 +1,164 @@
+package channels
+
+import (
+	"context"
+	"errors"
+	"os"
+	"testing"
+
+	"github.com/prometheus/alertmanager/notify"
+	"github.com/prometheus/alertmanager/template"
+	"github.com/prometheus/alertmanager/types"
+	"github.com/prometheus/common/model"
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/grafana/pkg/components/simplejson"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/alerting"
+)
+
+func TestPagerdutyNotifier(t *testing.T) {
+	tmpl, err := template.FromGlobs("templates/default.tmpl")
+	require.NoError(t, err)
+
+	hostname, err := os.Hostname()
+	require.NoError(t, err)
+
+	cases := []struct {
+		name         string
```

Comment:
There's one more interesting expectation you can test. With the bus, you can actually intercept messages sent through it - in this case, the bus command takes care of delivering the final result which works perfectly for testing.

Something like:
```
r := map[string]string{}
bus.AddHandler("test", func(webhook * models.SendWebhookSync) error {
  r["message"] = webhook.Body // repeat for other fields
  return nil
})

require.JSONeq(t, c.expectedWebhookBody, r["message"])
```

I like these types of tests becare they're very easy to understand (given it's just a JSON output) and makes changes dead simple w/o having to understand the current struct.